src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -0,0 +1,252 @@
+﻿namespace NServiceBus.Core.Tests.Msmq
+{
+    using System;
+    using System.Messaging;
+    using System.Security.Principal;
+    using NUnit.Framework;
+    using Transport;
+
+    [TestFixture]
+    public class MsmqQueueCreatorTests
+    {
+        [Test]
+        public void Should_create_all_queues()
+        {
+            var creator = new MsmqQueueCreator(true);
+            var bindings = new QueueBindings();
+
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.receiver");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.endpoint1");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.endpoint2");
+
+            bindings.BindReceiving("MsmqQueueCreatorTests.receiver");
+            bindings.BindSending("MsmqQueueCreatorTests.target1");
+            bindings.BindSending("MsmqQueueCreatorTests.target2");
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+            Assert.True(QueueExists("MsmqQueueCreatorTests.receiver"));
+            Assert.True(QueueExists("MsmqQueueCreatorTests.target1"));
+            Assert.True(QueueExists("MsmqQueueCreatorTests.target2"));
+        }
+
+        [Test]
+        public void Should_not_create_remote_queues()
+        {
+            var creator = new MsmqQueueCreator(true);
+            var bindings = new QueueBindings();
+            var remoteQueueName = "MsmqQueueCreatorTests.remote";
+
+            bindings.BindSending($"{remoteQueueName}@some-machine");
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+            Assert.False(QueueExists(remoteQueueName));
+        }
+
+
+        [Test]
+        public void Should_setup_permissions()
+        {
+            var testQueueName = "MsmqQueueCreatorTests.permissions";
+
+            var creator = new MsmqQueueCreator(true);
```

Comment:
In some places we use useTransactionalQueues: and in some places we don't. Can we align and use it everywhere?